types/validator_set.go

```diff
@@ -927,11 +927,13 @@ func (vals *ValidatorSet) ToProto() (*tmproto.ValidatorSet, error) {
 	}
 	vp.Validators = valsProto
 
+	// if vals.Proposer != nil {
```

Comment:
remove?

Reply:
also do we know that vals.Proposer will never be nil?

Reply:
this is meant to be removed. If a validatorSet is populated then it should always have a proposer.

Reply:
Just a heads up though `IsNilOrEmpty()` doesn't check that

Reply:
yup, if the proposer is missing it will error out as nil validator. If the proposer is missing from a validator set there may be a larger problem/ the software was modified.